graveler/committed/sstable/batch.go

```diff
@@ -0,0 +1,68 @@
+package sstable
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/treeverse/lakefs/graveler/committed"
+
+	"go.uber.org/atomic"
+)
+
+type BatchCloser struct {
+	results []committed.WriteResult
+	err     atomic.Error
```

Comment:
Not sure, the package seems to prefer [`*Error`](https://godoc.org/go.uber.org/atomic#NewError).  In particular I am not sure you want what would happen to `err` if I copied a `BatchCloser`.  `Error` holds a (`sync/atomic`) [`Value`](https://golang.org/pkg/sync/atomic/#Value), which "must not be copied after first use".

Reply:
I reckon we should get rid of atomics here, and possibly throughout the file.  I am not sure that they are worth the effort and the weird semantics that modern architectures + Go + Uber combine to put on them.